server/sonar-server/src/main/java/org/sonar/server/measure/ws/MeasureDtoToWsMeasure.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * SonarQube :: Server
+ * Copyright (C) 2009-2016 SonarSource SA
+ * mailto:contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.server.measure.ws;
+
+import org.sonar.api.measures.Metric;
+import org.sonar.db.measure.MeasureDto;
+import org.sonar.db.metric.MetricDto;
+import org.sonarqube.ws.WsMeasures;
+
+class MeasureDtoToWsMeasure {
+  private MeasureDtoToWsMeasure() {
+    // static methods
+  }
+
+  static WsMeasures.Measure measureDtoToWsMeasure(MetricDto metricDto, MeasureDto measureDto) {
+    WsMeasures.Measure.Builder measure = WsMeasures.Measure.newBuilder();
+    measure.setMetricKey(metricDto.getKey());
+    measure.setValue(measureValue(measureDto, metricDto));
+
+    if (measureDto.getVariation(1) != null) {
+      measure.setValueP1(measureVariationValue(measureDto.getVariation(1), metricDto));
+    }
+    if (measureDto.getVariation(2) != null) {
+      measure.setValueP2(measureVariationValue(measureDto.getVariation(2), metricDto));
+    }
+    if (measureDto.getVariation(3) != null) {
+      measure.setValueP3(measureVariationValue(measureDto.getVariation(3), metricDto));
+    }
+    if (measureDto.getVariation(4) != null) {
+      measure.setValueP4(measureVariationValue(measureDto.getVariation(4), metricDto));
+    }
+    if (measureDto.getVariation(5) != null) {
+      measure.setValueP5(measureVariationValue(measureDto.getVariation(5), metricDto));
+    }
+
+    return measure.build();
+  }
+
+  private static String measureValue(MeasureDto measure, MetricDto metric) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "measureValue" is 19 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)